jujugui/static/gui/src/app/utils/component-test-utils.js

```diff
@@ -318,5 +330,43 @@ var jsTestUtils = {
         }
       }
     };
+  },
+
+  /**
+    While assert.deepEqual is great, if a nested child errors it's
+    hard to debug. This function goes to the deepest level and works
+    it's way out - giving an easier to debug error.
+    Remember to bind the scope: 
+      jsTestUtils.specificDeepEquals.bind(this, output, expected);
+  */
+  specificDeepEquals: function(output, expected) {
```

Comment:
This is awesome! Plz send out an email to peeps (see us for the email) about this one